test/hotspot/jtreg/compiler/c1/TestCheckIndexC1Intrinsic.java

```diff
@@ -0,0 +1,166 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @author Yi Yang
+ * @bug 8265518
+ * @summary C1 intrinsic support for jdk.internal.util.Preconditions.checkIndex
+ * @requires vm.compiler1.enabled
+ * @library /test/lib
+ * @modules java.base/jdk.internal.util
+ * @run main/othervm -XX:TieredStopAtLevel=1 -Xbatch
+ *                   -XX:CompileCommand=dontinline,*TestCheckIndexC1Intrinsic.check*
+ *                   -XX:CompileCommand=compileonly,*TestCheckIndexC1Intrinsic.check*
+ *                   compiler.c1.TestCheckIndexC1Intrinsic
+ */
+
+package compiler.c1;
+
+import jdk.test.lib.Asserts;
+import jdk.internal.util.Preconditions;
+
+public class TestCheckIndexC1Intrinsic {
+    static int limit = 100;
+
+    private static class MyException extends RuntimeException {
+        public MyException(String msg) {
+            super(msg);
+        }
+    }
+
+    static void check0() {
+        long res = Preconditions.checkIndex(0, 1, null);
+        Asserts.assertEquals((int)res, 0);
+        try {
+            Preconditions.checkIndex(1, 1, null);
+            throw new AssertionError("Expected IndexOutOfBoundsException not thrown");
+        } catch (IndexOutOfBoundsException e) {
+            // got it!
+        }
+        try {
+            Preconditions.checkIndex(Integer.MIN_VALUE, -1, null);
+            throw new AssertionError("Expected IndexOutOfBoundsException not thrown");
+        } catch (IndexOutOfBoundsException e) {
+            // got it!
+        }
+        try {
+            Preconditions.checkIndex(Integer.MIN_VALUE, Integer.MIN_VALUE, null);
+            throw new AssertionError("Expected IndexOutOfBoundsException not thrown");
+        } catch (IndexOutOfBoundsException e) {
+            // got it!
+        }
+        try {
+            Preconditions.checkIndex(Long.MAX_VALUE, Long.MAX_VALUE, null);
+            throw new AssertionError("Expected IndexOutOfBoundsException not thrown");
+        } catch (IndexOutOfBoundsException e) {
+            // got it!
+        }
+        res = Preconditions.checkIndex(Long.MAX_VALUE - 1, Long.MAX_VALUE, null);
+        Asserts.assertEquals(res, Long.MAX_VALUE - 1);
+
+        try {
+            // read fields
+            Preconditions.checkIndex(limit + 1, limit, (s, integers) -> new MyException("Reason:" + s + "::" + integers));
+            throw new AssertionError("Expected IndexOutOfBoundsException not thrown");
+        } catch(MyException e){
+            // got it!
+        }
+    }
+
+    static void check1(int i) {
+        try {
+            Preconditions.checkIndex(i, 9999, (s, integers) -> new RuntimeException("ex"));
```

Comment:
I believe 
```
throw new AssertionError("Expected IndexOutOfBoundsException not thrown");
```

should be added in `check1`...`check4` as well...

Reply:
Hmm.. check1 and check2 should add. check3 and check4 are not necessary since the desired exception is thrown only if i==9999.

Reply:
Hmm.. They would throw desired exceptions only if i==9999.